flink-ml-lib/src/test/java/org/apache/flink/ml/classification/NaiveBayesTest.java

```diff
@@ -0,0 +1,297 @@
+                        .build();
+
+        trainData =
+                new Row[] {
+                    Row.of(1., Vectors.dense(1, 1., 1., 1., 2.), 11.0),
+                    Row.of(1., Vectors.dense(1, 1., 0., 1., 2.), 11.0),
+                    Row.of(1., Vectors.dense(2, 0., 1., 1., 3.), 11.0),
+                    Row.of(1., Vectors.dense(2, 0., 1., 1.5, 2.), 11.0),
+                    Row.of(2., Vectors.dense(3, 1.5, 1., 0.5, 3.), 10.0),
+                    Row.of(1., Vectors.dense(1, 1., 1.5, 0., 1.), 10.0),
+                    Row.of(2., Vectors.dense(4, 1., 1., 0., 1.), 10.0)
+                };
+
+        predictData = trainData;
+
+        expectedOutput =
+                new Row[] {
+                    Row.of(1., Vectors.dense(1, 1., 1., 1., 2.), 11.0, 11.0),
+                    Row.of(1., Vectors.dense(1, 1., 0., 1., 2.), 11.0, 11.0),
+                    Row.of(1., Vectors.dense(2, 0., 1., 1., 3.), 11.0, 11.0),
+                    Row.of(1., Vectors.dense(2, 0., 1., 1.5, 2.), 11.0, 11.0),
+                    Row.of(2., Vectors.dense(3, 1.5, 1., 0.5, 3.), 10.0, 10.0),
+                    Row.of(1., Vectors.dense(1, 1., 1.5, 0., 1.), 10.0, 10.0),
+                    Row.of(2., Vectors.dense(4, 1., 1., 0., 1.), 10.0, 10.0)
+                };
+
+        featuresCol = "features";
+        featureSize = 5;
+        labelCol = "label";
+        predictCol = "predict";
+        modelType = "categorical";
+        smoothing = 1.0;
+        isSaveLoad = false;
+    }
+
+    @Test
+    public void testNaiveBayes() throws Exception {
+        errorMessage = "normal test for Naive Bayes";
+        runAndCheck();
+    }
+
+    @Test(expected = Exception.class)
+    public void testInputNotContainFeature() throws Exception {
+        errorMessage =
+                "Naive Bayes should throw exception if some feature columns are missing from train data";
+        featuresCol = "feature2";
+        runAndCheck();
+    }
+
+    @Test(expected = Exception.class)
+    public void testPredictUnseenFeature() throws Exception {
+        errorMessage =
+                "Naive Bayes should throw exception if unseen feature values are met in prediction "
+                        + "and the model type is categorical.";
+        predictData =
+                new Row[] {
+                    Row.of(1., Vectors.dense(5, 1., 1., 1., 2.), 11.0),
+                    Row.of(1., Vectors.dense(5, 1., 0., 1., 2.), 11.0),
+                    Row.of(1., Vectors.dense(2, 0., 1., 1., 3.), 11.0),
+                    Row.of(1., Vectors.dense(2, 0., 1., 1.5, 2.), 11.0),
+                    Row.of(2., Vectors.dense(3, 1.5, 1., 0.5, 3.), 10.0),
+                    Row.of(1., Vectors.dense(1, 1., 1.5, 0., 1.), 10.0),
+                    Row.of(2., Vectors.dense(4, 1., 1., 0., 1.), 10.0)
+                };
+        runAndCheck();
+    }
+
+    @Test(expected = Exception.class)
+    public void testEmptyLabel() throws Exception {
```

Comment:
It seems a bit verbose to have a dedicated test for each param. We probably don't want to do this for every parameter of every Estimator/Transformer.

And the purpose of this test seems to have already been covered by `StageTest::testValidators`. We can add unit tests there if needed.